### PR TITLE
Split requirements.txt into per-discipline files and add per-directory READMEs

### LINKED ISSUE #90: Split requirements.txt into per-discipline files and add per-directory READMEs
## Summary

Currently all dependencies for all notebooks are listed in a single `requirements.txt`. Users who only want to run one or two notebooks must install heavyweight dependencies they don't need (e.g., `spiceypy`, `pywwt`, `obspy`, `geopandas`).

**Proposed solution:** Add per-discipline/group `requirements-<group>.txt` files so users can install only what they need. Update the top-level README and add per-directory READMEs with install instructions.

## Requirements Files to Create

| File | Notebooks Covered | Key Packages |
|------|-------------------|--------------|
| `requirements-geo.txt` | `discipline/geo/` | astropy, matplotlib, numpy, Pillow, pds4-tools, pdr, pds.peppi, psutil, requests |
| `requirements-img.txt` | `discipline/img/` | geopandas, ipyleaflet, leafmap, numpy, pandas, pds.peppi, pds4-tools, requests |
| `requirements-ppi.txt` | `discipline/ppi/` | astropy, matplotlib, numpy, pandas, pds.peppi, pds4-tools, pywwt, requests, spiceypy |
| `requirements-sbn.txt` | `discipline/sbn/` | astropy, matplotlib, pds.api_client, pds.peppi, pdr, requests, tqdm |
| `requirements-peppi-advanced.txt` | `generic/peppi-advanced/` | matplotlib, pandas, pds.peppi |
| `requirements-training.txt` | `generic/training_20260305/` | pandas, pds.peppi, requests |
| `requirements-simple.txt` | `generic/simple/` | pandas, pdr, pds.peppi |

## README Updates

- Top-level `README.md`: add section explaining per-group install approach alongside full install
- New `notebooks/discipline/geo/README.md`: notebook descriptions + `pip install -r requirements-geo.txt`
- New `notebooks/discipline/img/README.md`: notebook descriptions + `pip install -r requirements-img.txt`
- New `notebooks/discipline/ppi/README.md`: notebook descriptions + `pip install -r requirements-ppi.txt`
- New `notebooks/discipline/sbn/README.md`: notebook descriptions + `pip install -r requirements-sbn.txt`
- New `notebooks/generic/simple/README.md`: notebook descriptions + `pip install -r requirements-simple.txt`
- New `notebooks/generic/training_20260305/README.md`: notebook descriptions + `pip install -r requirements-training.txt`
- Update `notebooks/generic/peppi-advanced/README.md`: reference `requirements-peppi-advanced.txt`

## Acceptance Criteria

- [ ] Per-group requirements files created at the repo root
- [ ] Each requirements file only lists packages needed for that group's notebooks
- [ ] Top-level README updated with per-group install instructions
- [ ] Per-directory READMEs created/updated with install instructions and notebook descriptions

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Splits the monolithic `requirements.txt` into seven per-discipline requirements files so users can install only the packages needed for the notebooks they want to run. Adds per-directory README files with install instructions and notebook descriptions.

**Changes:**
- Added 7 per-discipline requirements files at the repo root: `requirements-geo.txt`, `requirements-img.txt`, `requirements-ppi.txt`, `requirements-sbn.txt`, `requirements-peppi-advanced.txt`, `requirements-training.txt`, `requirements-simple.txt`
- Updated top-level `README.md` with an Option A (install everything) / Option B (install per-discipline) guide including a quick-reference table
- Added `README.md` for: `notebooks/discipline/geo/`, `notebooks/discipline/img/`, `notebooks/discipline/ppi/`, `notebooks/discipline/sbn/`, `notebooks/generic/simple/`, `notebooks/generic/training_20260305/`
- Updated `notebooks/generic/peppi-advanced/README.md` to reference `requirements-peppi-advanced.txt` and corrected Python version requirement (3.13 → 3.12)

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90%

## ⚙️ Test Data and/or Report

This PR contains only requirements files and documentation (no executable code changes). Validation steps:
1. `pip install -r requirements-peppi-advanced.txt` and confirm notebooks in `notebooks/generic/peppi-advanced/` run without import errors
2. `pip install -r requirements-geo.txt` and confirm notebooks in `notebooks/discipline/geo/` run without import errors
3. Verify README renders correctly on GitHub

## ♻️ Related Issues

Resolves #90
Refs #89

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: notebooks/discipline/geo/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Geosciences Node Notebooks
+
+Notebooks demonstrating PDS API access for geosciences data, including radio science, digital elevation models, and planetary flyby visualizations.
+
+## Installation
+
+```bash
+pip install -r requirements-geo.txt
```
Comment (nutjob4life) [inline notebooks/discipline/geo/README.md]: Inside of documentation I like to see the `--long` formats as they're "self-documenting":
```bash
pip install --requirement requirements-geo.txt
```

File: notebooks/discipline/ppi/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Run from the repository root. Then start JupyterLab:
+
+```bash
+jupyter-lab
+```
+
+> **Note:** The `GLL_MagPlot` notebook uses pyWWT for 3D visualization. After installing, you may need to enable the JupyterLab extension:
+> ```bash
+> jupyter labextension install @wwtelescope/jupyterlab
```
Comment (nutjob4life) [inline notebooks/discipline/ppi/README.md]: Is this step necessary? This PR also adds `wwt-jupyterlab-extension` to `requirements-ppi.txt` which may be sufficient.

File: README.md [hunk 3]
```diff
@@ -20,23 +20,46 @@ Clone the current repository:
...
+```
+
+**Option A — Install everything** (all notebooks, all dependencies):
+
+```console
 $ pip install --requirement requirements.txt
 ```
 
+**Option B — Install only what you need** (faster, fewer packages):
+
```
Comment (nutjob4life) [inline README.md:38]: Under Option B the `jupyterlab` command may not be available as most of the new per-directory `requirements-*.txt` files don't list it explicitly as a dependency. Either they should or we should mention a separate `pip install jupyterlab` step.

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Just some tiny little nits; what do you think?

File: README.md (hunk 3 above)
Comment (jordanpadams) [reply README.md:38]: added here: https://github.com/NASA-PDS/search-api-notebook/pull/91/changes#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R60

File: notebooks/discipline/ppi/README.md (hunk 2 above)
Comment (jordanpadams) [reply notebooks/discipline/ppi/README.md]: resolved per https://github.com/NASA-PDS/search-api-notebook/pull/91/commits/35e7417a545a50c4d6ccb3c64af6f9c64e6ac1be

File: notebooks/discipline/geo/README.md (hunk 1 above)
Comment (jordanpadams) [reply notebooks/discipline/geo/README.md]: resolved per https://github.com/NASA-PDS/search-api-notebook/pull/91/commits/35e7417a545a50c4d6ccb3c64af6f9c64e6ac1be

**Review by nutjob4life — APPROVED** [review]
Fabulous @jordanpadams! Thanks for putting up with my pedantic nits 🤓
